### DISCUSSION TITLE
pip install prophet error

### DISCUSSION THREAD

**Discussion by alexisgonzalez507** [community/community#69369] [Programming Help]
### Body

How do I fix this error I'm encountering while installing Prophet? I have Python 3.11, and I tried it with Python 3.12, and I'm getting the same errors

1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for ephem
Failed to build prophet ephem
ERROR: Could not build wheels for prophet, ephem, which is required to install pyproject.toml-based projects

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by MadhushaPrasad** [comment]
I think the error is when you're encountering a issue during the installation of the Prophet library, particularly with the 'ephem' dependency. Here are some steps that you can take to resolve this issue:

- Update System:
    Make sure your system is up to date using ``sudo apt-get`` update and ``sudo apt-get upgrade.``
- Install Dependencies:
    Install ``libffi-dev`` with ``sudo apt-get install libffi-dev.``
- Python Compatibility:
    Check if Prophet is compatible with ``Python 3.11/3.12``. Try ``Python 3.9/3.10.`` , I think most suitable version is the v3.9 try it.Because why I am saying like this, I got some errors during installation of some libs in v3.11 and its fixed after moving to v3.9, so go & try that version.
- Compiler Compatibility:
    Ensure a compatible C compiler with ``sudo apt-get install build-essential``.
- Use Virtual Environment:
    Create a virtual environment and install Prophet within it some it will be fixed inside the v.environment.
- Specify Versions:
    Try installing specific versions of ephem and prophet.
   
If you have not referred to the official documentation for any specific instructions that will also help solve this issue, they mention some versions that can be installed in the minimum supported Python version.like below

```

    From v0.6 onwards, Python 2 is no longer supported.
    As of v1.0, the package name on PyPI is “prophet”; prior to v1.0 it was “fbprophet”.
    As of v1.1, the minimum supported Python version is 3.7.
```
 [here is the python installing guide for prophet](https://facebook.github.io/prophet/docs/installation.html#python)
 After installation, you can [get started! from this link](https://facebook.github.io/prophet/docs/quick_start.html#r-api)

**Comment by alexisgonzalez507** [comment]
[alexisgonzalez@AirM2deAlexis ~ % python3.9 -m pip install prophet

Collecting prophet
  Using cached prophet-1.1.4.tar.gz (68 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting cmdstanpy>=1.0.4 (from prophet)
  Obtaining dependency information for cmdstanpy>=1.0.4 from https://files.pythonhosted.org/packages/e8/e2/2ddc273e726680a3c816f9a05c6726d57d61820fa7ab4753ed282d0a8bbe/cmdstanpy-1.2.0-py3-none-any.whl.metadata
  Using cached cmdstanpy-1.2.0-py3-none-any.whl.metadata (3.9 kB)
Requirement already satisfied: numpy>=1.15.4 in /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages (from prophet) (1.26.0)
Collecting matplotlib>=2.0.0 (from prophet)
  Obtaining dependency information for matplotlib>=2.0.0 from https://files.pythonhosted.org/packages/95/25/4178ff92b6e523c9ad48ed3822b26375d5a5173b28b75ac7c1cb640a9c94/matplotlib-3.8.0-cp39-cp39-macosx_11_0_arm64.whl.metadata
  Using cached matplotlib-3.8.0-cp39-cp39-macosx_11_0_arm64.whl.metadata (5.8 kB)
Collecting pandas>=1.0.4 (from prophet)
  Obtaining dependency information for pandas>=1.0.4 from https://files.pythonhosted.org/packages/c5/2f/bf85305b044ddee0ade62c444c7ef551eb423899424b3898d60895d02f63/pandas-2.1.1-cp39-cp39-macosx_11_0_arm64.whl.metadata
  Using cached pandas-2.1.1-cp39-cp39-macosx_11_0_arm64.whl.metadata (18 kB)
Collecting LunarCalendar>=0.0.9 (from prophet)
  Using cached LunarCalendar-0.0.9-py2.py3-none-any.whl (18 kB)
Collecting convertdate>=2.1.2 (from prophet)
  Using cached convertdate-2.4.0-py3-none-any.whl (47 kB)
Collecting holidays>=0.25 (from prophet)
  Obtaining dependency information for holidays>=0.25 from https://files.pythonhosted.org/packages/2b/21/12a180f7dc75758d466ff1eeaef894504320d9af4fd48d50080129f5538d/holidays-0.34-py3-none-any.whl.metadata
  Using cached holidays-0.34-py3-none-any.whl.metadata (17 kB)
Requirement already satisfied: python-dateutil>=2.8.0 in ./Library/Python/3.9/lib/python/site-packages (from prophet) (2.8.2)
Collecting tqdm>=4.36.1 (from prophet)
  Obtaining dependency information for tqdm>=4.36.1 from https://files.pythonhosted.org/packages/00/e5/f12a80907d0884e6dff9c16d0c0114d81b8cd07dc3ae54c5e962cc83037e/tqdm-4.66.1-py3-none-any.whl.metadata
  Using cached tqdm-4.66.1-py3-none-any.whl.metadata (57 kB)
Collecting importlib-resources (from prophet)
  Obtaining dependency information for importlib-resources from https://files.pythonhosted.org/packages/65/6e/09d8816b5cb7a4006ef8ad1717a2703ad9f331dae9717d9f22488a2d6469/importlib_resources-6.1.0-py3-none-any.whl.metadata
  Using cached importlib_resources-6.1.0-py3-none-any.whl.metadata (4.1 kB)
Collecting stanio~=0.3.0 (from cmdstanpy>=1.0.4->prophet)
  Obtaining dependency information for stanio~=0.3.0 from https://files.pythonhosted.org/packages/26/01/e2579afa3ad1eade8b54bf5b9e45790d219681b1f8f97edbd2f18c4e1a7d/stanio-0.3.0-py3-none-any.whl.metadata
  Using cached stanio-0.3.0-py3-none-any.whl.metadata (963 bytes)
Collecting pymeeus<=1,>=0.3.13 (from convertdate>=2.1.2->prophet)
  Using cached PyMeeus-0.5.12-py3-none-any.whl
Collecting ephem>=3.7.5.3 (from LunarCalendar>=0.0.9->prophet)
  Using cached ephem-4.1.4-cp39-cp39-macosx_10_9_universal2.whl
Collecting pytz (from LunarCalendar>=0.0.9->prophet)
  Obtaining dependency information for pytz from https://files.pythonhosted.org/packages/32/4d/aaf7eff5deb402fd9a24a1449a8119f00d74ae9c2efa79f8ef9994261fc2/pytz-2023.3.post1-py2.py3-none-any.whl.metadata
  Using cached pytz-2023.3.post1-py2.py3-none-any.whl.metadata (22 kB)
Collecting contourpy>=1.0.1 (from matplotlib>=2.0.0->prophet)
  Obtaining dependency information for contourpy>=1.0.1 from https://files.pythonhosted.org/packages/09/fe/086e6847ee53da10ddf0b6c5e5f877ab43e68e355d2f4c85f67561ee8a57/contourpy-1.1.1-cp39-cp39-macosx_11_0_arm64.whl.metadata
  Using cached contourpy-1.1.1-cp39-cp39-macosx_11_0_arm64.whl.metadata (5.9 kB)
Collecting cycler>=0.10 (from matplotlib>=2.0.0->prophet)
  Obtaining dependency information for cycler>=0.10 from https://files.pythonhosted.org/packages/2b/b3/70c33027c4918c10ccf176014b38f8b91cb18ac018a78854543a4fc72609/cycler-0.12.0-py3-none-any.whl.metadata
  Using cached cycler-0.12.0-py3-none-any.whl.metadata (3.8 kB)
Collecting fonttools>=4.22.0 (from matplotlib>=2.0.0->prophet)
  Obtaining dependency information for fonttools>=4.22.0 from https://files.pythonhosted.org/packages/e8/85/580c7212c0b12e155966268e0dc0a84479900cab29c73eca27e2533c9fee/fonttools-4.43.0-cp39-cp39-macosx_10_9_universal2.whl.metadata
  Using cached fonttools-4.43.0-cp39-cp39-macosx_10_9_universal2.whl.metadata (151 kB)
Collecting kiwisolver>=1.0.1 (from matplotlib>=2.0.0->prophet)
  Obtaining dependency information for kiwisolver>=1.0.1 from https://files.pythonhosted.org/packages/89/a8/3b7e14121bea4438b87630557645bb7648b17b54acaa39b93f4bf7f8d33e/kiwisolver-1.4.5-cp39-cp39-macosx_11_0_arm64.whl.metadata
  Using cached kiwisolver-1.4.5-cp39-cp39-macosx_11_0_arm64.whl.metadata (6.4 kB)
Requirement already satisfied: packaging>=20.0 in ./Library/Python/3.9/lib/python/site-packages (from matplotlib>=2.0.0->prophet) (23.2)
Collecting pillow>=6.2.0 (from matplotlib>=2.0.0->prophet)
  Obtaining dependency information for pillow>=6.2.0 from https://files.pythonhosted.org/packages/3c/56/8449d670bae6e8f1b45e12cc2746561e42d3d80e3a432bf6f6e71c260aa1/Pillow-10.0.1-cp39-cp39-macosx_11_0_arm64.whl.metadata
  Using cached Pillow-10.0.1-cp39-cp39-macosx_11_0_arm64.whl.metadata (9.5 kB)
Collecting pyparsing>=2.3.1 (from matplotlib>=2.0.0->prophet)
  Obtaining dependency information for pyparsing>=2.3.1 from https://files.pythonhosted.org/packages/39/92/8486ede85fcc088f1b3dba4ce92dd29d126fd96b0008ea213167940a2475/pyparsing-3.1.1-py3-none-any.whl.metadata
  Using cached pyparsing-3.1.1-py3-none-any.whl.metadata (5.1 kB)
Requirement already satisfied: zipp>=3.1.0 in ./Library/Python/3.9/lib/python/site-packages (from importlib-resources->prophet) (3.17.0)
Collecting tzdata>=2022.1 (from pandas>=1.0.4->prophet)
  Using cached tzdata-2023.3-py2.py3-none-any.whl (341 kB)
Collecting six>=1.5 (from python-dateutil>=2.8.0->prophet)
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Using cached cmdstanpy-1.2.0-py3-none-any.whl (93 kB)
Using cached holidays-0.34-py3-none-any.whl (778 kB)
Using cached matplotlib-3.8.0-cp39-cp39-macosx_11_0_arm64.whl (7.5 MB)
Using cached importlib_resources-6.1.0-py3-none-any.whl (33 kB)
Using cached pandas-2.1.1-cp39-cp39-macosx_11_0_arm64.whl (10.9 MB)
Using cached tqdm-4.66.1-py3-none-any.whl (78 kB)
Using cached contourpy-1.1.1-cp39-cp39-macosx_11_0_arm64.whl (232 kB)
Using cached cycler-0.12.0-py3-none-any.whl (8.2 kB)
Using cached fonttools-4.43.0-cp39-cp39-macosx_10_9_universal2.whl (2.7 MB)
Using cached kiwisolver-1.4.5-cp39-cp39-macosx_11_0_arm64.whl (66 kB)
Using cached Pillow-10.0.1-cp39-cp39-macosx_11_0_arm64.whl (3.3 MB)
Using cached pyparsing-3.1.1-py3-none-any.whl (103 kB)
Using cached pytz-2023.3.post1-py2.py3-none-any.whl (502 kB)
Using cached stanio-0.3.0-py3-none-any.whl (6.2 kB)
Building wheels for collected packages: prophet
  Building wheel for prophet (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for prophet (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [68 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-10.9-universal2-cpython-39
      creating build/lib.macosx-10.9-universal2-cpython-39/prophet
      creating build/lib.macosx-10.9-universal2-cpython-39/prophet/stan_model
      22:39:57 - cmdstanpy - WARNING - CmdStan installation failed.
      Version 2.31.0 cannot be downloaded. Connection to GitHub failed. Check firewall settings or ensure this version exists.
      CmdStan installation failed.
      Version 2.31.0 cannot be downloaded. Connection to GitHub failed. Check firewall settings or ensure this version exists.
      Compiling cmdstanpy model
      Installing cmdstan to /private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-install-46wasve8/prophet_31d1f42360534a4baaf7dedc918634e3/build/lib.macosx-10.9-universal2-cpython-39/prophet/stan_model/cmdstan-2.31.0
      CmdStan install directory: /private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-install-46wasve8/prophet_31d1f42360534a4baaf7dedc918634e3/build/lib.macosx-10.9-universal2-cpython-39/prophet/stan_model
      checking version 2.31.0 availability, retry (1/5)
      checking version 2.31.0 availability, retry (2/5)
      checking version 2.31.0 availability, retry (3/5)
      checking version 2.31.0 availability, retry (4/5)
      checking version 2.31.0 availability, retry (5/5)
      Release 2.31.0 is unavailable from URL https://github.com/stan-dev/cmdstan/releases/download/v2.31.0/cmdstan-2.31.0.tar.gz
      URLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)
      Traceback (most recent call last):
        File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 251, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 434, in build_wheel
          return self._build_with_temp_dir(
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 419, in _build_with_temp_dir
          self.run_setup()
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 341, in run_setup
          exec(code, locals())
        File "<string>", line 200, in <module>
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/__init__.py", line 103, in setup
          return distutils.core.setup(**attrs)
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/dist.py", line 989, in run_command
          super().run_command(command)
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/wheel/bdist_wheel.py", line 364, in run
          self.run_command("build")
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/dist.py", line 989, in run_command
          super().run_command(command)
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/_distutils/command/build.py", line 131, in run
          self.run_command(cmd_name)
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/dist.py", line 989, in run_command
          super().run_command(command)
        File "/private/var/folders/qv/zjx7jm593vx6kdfmpv492nzc0000gn/T/pip-build-env-qeal8acv/overlay/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "<string>", line 161, in run
        File "<string>", line 148, in build_models
        File "<string>", line 122, in build_cmdstan_model
        File "<string>", line 99, in install_cmdstan_deps
      RuntimeError: CmdStan failed to install in repackaged directory
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for prophet
Failed to build prophet
ERROR: Could not build wheels for prophet, which is required to install pyproject.toml-based projects](url)

**Reply by alexisgonzalez507** [reply]
This is the installation message, and if you see, there are several errors. I tried using another version of Python 3.9 and followed the recommendations, but it didn't work (I'm using a MacBook Air M2).
